src/test/java/com/hotline/tests/registrationpage/RegistrationNegativeTest.java

```diff
@@ -0,0 +1,104 @@
+package com.hotline.tests.registrationpage;
+
+import com.hotline.helpers.EErrors;
+import com.hotline.helpers.ReusableMethods;
+import com.hotline.pageobject.pages.RegistrationFinalPage;
+import com.hotline.pageobject.pages.RegistrationPage;
+import com.hotline.tests.BaseTest;
+import com.hotline.user.User;
+import com.hotline.user.UserInfo;
+import io.qameta.allure.Allure;
+import io.qameta.allure.Description;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.hotline.helpers.EErrors.*;
+
+public class RegistrationNegativeTest extends BaseTest {
+    @Description("Verify that we can't register user with invalid credentials")
+    @Parameters({"User register; List<EErrors> expectedErrors"})
+    @Test(dataProvider = "invalidUsers")
+    public void registerInvalidUser(User user, List<EErrors> expectedErrors) {
+        Allure.step("Start test registration user with invalid credentials");
+        RegistrationPage registrationPage = openBrowser().gotoLoginPage().gotoRegistrationPage();
+
+        registrationPage.getLoginField().sendKeys(user.getLogin());
+        registrationPage.getNameField().sendKeys(user.getName());
+        registrationPage.getPasswordField().sendKeys(user.getPassword());
+        registrationPage.getShowPasswordBtn().click();
```

Comment:
Why do you click on show password button here?

Reply:
to see that I can see my password and it current. I see, i can also add here password assert if it's the same one.)

Reply:
There is no real need to that, you can expose a password in such way. A better solution can be create a separate test for this button